realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -157,8 +158,11 @@ void BaseRealSenseNode::setupFilters()
     _filters.push_back(std::make_shared<NamedFilter>(std::make_shared<rs2::temporal_filter>(), _parameters, _logger));
     _filters.push_back(std::make_shared<NamedFilter>(std::make_shared<rs2::hole_filling_filter>(), _parameters, _logger));
     _filters.push_back(std::make_shared<NamedFilter>(std::make_shared<rs2::disparity_transform>(false), _parameters, _logger));
-    _align_depth_filter = std::make_shared<NamedFilter>(std::make_shared<rs2::align>(RS2_STREAM_COLOR), _parameters, _logger);
+
+    std::function<void(const rclcpp::Parameter&)> update_sensor_func = [this](const rclcpp::Parameter&){_is_align_depth_changed = true; _cv_mpc.notify_one();};
```

Comment:
I know it wasn't implemented like that on `is_profile_changed` but I think it was missed and the behavior can be different than expected sometimes.

When using conditional variables between threads the thread that change the shared variable has to lock (mutex) the variable once he is changing it with the same lock that used to lock the CV.

You can read about it here:
1. https://en.cppreference.com/w/cpp/thread/condition_variable
2. https://www.modernescpp.com/index.php/c-core-guidelines-be-aware-of-the-traps-of-condition-variables